### ISSUE TITLE
Android specific: Picker returns duplicate images (missing some images)in returned array from multi select photo gallery option

### ISSUE THREAD

**Issue by rajeshrthl** [ivpusic/react-native-image-crop-picker#687]
### Version
Tell us which versions you are using: 

- react-native-image-crop-picker v0.19.3
- react-native v0.51.0

### Platform
Tell us to which platform this issue is related

- Android

### Expected behaviour
Image Picker should return exact number of unique images in array when multiple images are selected from Android photo gallery. Images selected should be equal to images returned in promise then callback. There should not be any images missing in returned image array.

### Actual behaviour
Images array returned from multiple image picker option returns duplicate images in Android. In iOS this feature works fine. This issue is Android only and not consistent. Sometimes the picker returns exact images, sometimes 2-3 images are duplicate, sometimes 1 or 2. It has been tested with different number of images and different images from different location in phone gallery. Duplicate images come in array returned from multiselect picker from phone gallery. As a result of duplicate images some images are missing as compared to original selection from photo gallery.

### Steps to reproduce

1. Open picker with these settings:
ImagePicker.openPicker({
      multiple: true,
      waitAnimationEnd: false,
      includeBase64: true,
      maxFiles: 20,
      mediaType: "photo",
      compressImageQuality: 0.5
    })
      .then(selectedImages => {/*selectedImages array have duplicate images and as a result some images are missing from original selection*/})
.catch(e => {
        this.setState({ loadingPreview: false });
        console.log(e);
      });

2. In above then block do a checking to find duplicate entries or unique entries checking. Result of this checking shows that the array has duplicate entries. (use key as "data", "size" or "modificationDate" in lodash function - uniq or union any other function to check duplicate entries in selectedImages array)

3. Use includeBase64: true, as the faulty behaviour is noticed with this option. When original selection is compared with returned array there are some images missing in returned array.

### Attachments

// stacktrace or any other useful debug info

Love react-native-image-crop-picker? Please consider supporting our collective:
👉  https://opencollective.com/react-native-image-crop-picker/donate

**Comment by ivpusic** [comment]
can you investigate this?

**Comment by rajeshrthl** [comment]
@ivpusic this needs a fix in java classes as its native functionality in Android. Could you or anyone help me with this? This is really a show stopper for us as we need multiple image selection in our app. Thank you.

**Comment by superandrew213** [comment]
I'm seeing this too.

**Comment by ivpusic** [comment]
@superandrew213 do you have an idea why this might be happening?

**Comment by superandrew213** [comment]
@ivpusic not sure why it's happening. I can't see anything wrong with the code. I will do some more tests but it doesn't happen that often for me so it's a bit hard to test.

**Comment by superandrew213** [comment]
No unfortunately not. One image is replaced with a duplicate. So in the example above I selected 4 images but 1 is missing and is replaced with a duplicate of another image.

**Comment by rajeshrthl** [comment]
Thanks @ivpusic @superandrew213  for your time to resolve this issue. Appreciate your efforts. Hope you guys can find a solution soon.

**Comment by rajeshrthl** [comment]
Any update guys?

**Comment by superandrew213** [comment]
I haven't had much time to look into this.

I couldn't see anything wrong with code that could cause replacing some of the images with duplicates. But it's definitely happening though.

**Comment by rajeshrthl** [comment]
@superandrew213 yes its happening quiet often. Can you or someone have a look please. Any workaround you can think of?

**Comment by akiwarheit** [comment]
Investigating this as well.

What gallery app are you guys getting the duplicates? Google / Google Photos?

`Intent.ACTION_GET_CONTENT` should call your default Gallery app.

**Comment by superandrew213** [comment]
@akiwarheit I'm selecting from the phone (default) Gallery & Recents.

**Comment by sraka1** [comment]
@akiwarheit @ivpusic Did any of you find the root cause of this issue? I'm experiencing this as well and am trying to debug it.

**Comment by tres** [comment]
Hey friends, 

It would seem to me that maybe the issue as described could be explained by 
this:
https://github.com/ivpusic/react-native-image-crop-picker/blob/master/android/src/main/java/com/reactnative/ivpusic/imagepicker/PickerModule.java#L388
and this:
https://github.com/ivpusic/react-native-image-crop-picker/blob/master/android/src/main/java/com/reactnative/ivpusic/imagepicker/PickerModule.java#L402

Could it be that errors are silently being eaten and we just never see what's wrong when there's an error?

Wouldn't we expect or want that exception handled here instead:
https://github.com/ivpusic/react-native-image-crop-picker/blob/407ffdf837faa649162f1dd8d17b50870aea1a46/android/src/main/java/com/reactnative/ivpusic/imagepicker/PickerModule.java#L643


So if this theory is valid, what would be happening is that we have a notification back to the ResultCollector instance that's null. The ResultCollector is trying to use pushMap & that's causing some undefined behavior. 

There's a lot of assumptions in that & hopefully this helps add more signal than noise, but I think the base of the problem is that we're just eating exceptions silently.

Thanks for all your work!

**Comment by boaz-lachman** [comment]
any updates regarding this issue?
